dali/ft/filecopy.cpp

```diff
@@ -1219,19 +1210,66 @@ void FileSprayer::calculateSprayPartition()
     if ((partitioners.ordinality() > 0) && !srcAttr->hasProp("ECL"))
     {
         // Store discovered CSV record structure into target logical file.
-        StringBuffer recStru;
-        partitioners.item(0).getRecordStructure(recStru);
-        if ((recStru.length() > 0) && strstr(recStru.str(),"END;"))
-        {
-            if (distributedTarget)
-                distributedTarget->setECL(recStru.str());
-        }
+        StoreCsvRecordStructure(partitioners.item(0));
```

Comment:
Instead of these being here, it would be better if called after any calculate*partioner() call , i.e. spray()

Since it is not conditional on CSV here (always calls) and needn't be in any other case. Although CSV will be the only one that will will in recordStructure.

The partioner calls currently call ExamineCsvStructure (e.g. calculateOne2OnePartition) would setup a single partioner (for part 0) if CSV and therefore trigger this moved if ((partitioners.ordinality() > 0) && !srcAttr->hasProp("ECL")) check which sets the record structure into the file.